user-interface/0002-WCON-connect_vega_wallet.md

```diff
@@ -0,0 +1,76 @@
+# Connect Vega wallet & select keys
+
+## Connect wallet for the first time
+
+When looking to use Vega via a user interface e.g. Dapp (Decentralized web App), I...
+
+- **should** see a link to get a Vega wallet (in case I don't already have one) <a name="0002-WCON-036" href="#0002-WCON-036">0002-WCON-036</a>
+- **should** see a link to connect that opens up connection options <a name="0002-WCON-001" href="#0002-WCON-001">0002-WCON-001</a>
+- **must** select a connection method / wallet type: <a name="0002-WCON-002" href="#0002-WCON-002">0002-WCON-002</a>
+- if Rest:
+  - **must** have the option to input a non-default Wallet location <a name="0002-WCON-003" href="#0002-WCON-003">0002-WCON-003</a>
+  - **should** warn if the dapp is unable the see a wallet is running at the wallet location  <a name="0002-WCON-004" href="#0002-WCON-004">0002-WCON-004</a>
+  - **must** submit attempt to connect to wallet <a name="0002-WCON-005" href="#0002-WCON-005">0002-WCON-005</a>
+  - **could** trigger the app to open on the user's machine with a `vegawallet://` prompt <!--<a name="0002-WCON-006" href="#0002-WCON-006">0002-WCON-006</a>-->
+  
+  - if the dapp DOES already have a permission with the wallet: **must** see that wallet is connected <a name="0002-WCON-007" href="#0002-WCON-007">0002-WCON-007</a>
+    - if the app uses one key at a time: **should** show what key is active (re-use the last active key) <a name="0002-WCON-008" href="#0002-WCON-008">0002-WCON-008</a>
+
+  - if the wallet does NOT have an existing permission with the wallet: **must** prompt user to check wallet app to approve the request to connect wallet: See [Connecting to Dapps](0002-WCON-connect_vega_wallet.md#connect-wallet-for-the-first-time) for what should happen in wallet app <a name="0002-WCON-009" href="#0002-WCON-009">0002-WCON-009</a>
+  
+  - if new keys are given permission: **must** show the user the keys have been approved <a name="0002-WCON-010" href="#0002-WCON-010">0002-WCON-010</a>
+    - **should** see [public key(s)](7001-DATA-data_display.md#public-keys). <a name="0002-WCON-037" href="#0002-WCON-037">0002-WCON-037</a>
+    - **should** see alias(es) <a name="0002-WCON-011" href="#0002-WCON-011">0002-WCON-011</a>
+    - **could** see assets on key(s) <a name="0002-WCON-012" href="#0002-WCON-012">0002-WCON-012</a>
```

Comment:
This would be the dapp, fetching asset details from datanode for the keys that it has been given by the wallet.
To prevent an app leaking info about what keys are in a given wallet, I suggest we leave it to the user only granting permission for keys they are willing to (when that is possible). 
@barnabee Would you prefer this wasn't done until the user has the ability to control what keys are shared with the wallet?

Reply:
I'd prefer that **connecting to data node to get assets and balances isn't done at all** until the user can control whether or not they are willing to have wallet query datanode to fetch balances, with a clear understanding that doing so leaks their addresses to the datanode operator.

In fact I think that's serious enough that it is a blocker.

Once that is in place, if a user has approved getting the info, then showing it here doesn't seem like an issue.

Reply:
I've added an ac in https://github.com/vegaprotocol/specs/pull/1181/commits/9ab47aeab2b167ee147e7de4c27a4150fa7f3f54
> - **must** be able to control whether the wallet app queries for data (e.g. asset balances) on each key (to prevent info leaking about what keys belong to a wallet) <a name="0001-WALL-078" href="#0001-WALL-078">0001-WALL-078</a>

Reply:
We are commenting on a part of the file that applies to the dApp, in the context of connecting to a Vega wallet.
I think in this context it is okay to query a data node for all the keys I have given the app access too.
How do you feel about that @barnabee ?

Reply:
It's probably fine. I can imagine a dapp or app that uses the wallet to sign stuff where it wouldn't be ideal to have any leaking of data to the internet but I don't think we are planning to build any at the moment so I am resolving this.